modules/qualitycontrol/QualityController.ts

```diff
@@ -231,6 +266,80 @@ export default class QualityController {
         return false;
     }
 
+    /**
+     * Adjusts codec, lastN or receive resolution based on the send resolution (of the outbound streams) and limitation
+     * reported by the browser in the WebRTC stats. Recovery is also attempted if the limitation goes away. No action
+     * is taken if the adaptive mode has been disabled through config.js.
+     *
+     * @param {ISourceStats} sourceStats - The outbound-rtp stats for a local video track.
+     * @returns {void}
+     */
+    _performQualityOptimizations(sourceStats: ISourceStats): void {
+        // Do not attempt run time adjustments if the adaptive mode is disabled.
+        if (!this._enableAdaptiveMode) {
+            return;
+        }
+
+        const { encodeResolution, localTrack, qualityLimitationReason, tpc } = sourceStats;
+        const trackId = localTrack.rtcId.toString();
```

Comment:
isn't this a string already?

Reply:
Nope, its rtcTrackIdCounter that is passed when creating track and that is a number.